File: app/src/main/java/com/juawapps/newsspread/image/ImageLoaderWrapper.java
package com.juawapps.newsspread.image;

import android.support.v4.app.Fragment;
import android.widget.ImageView;

/**
 * This is a wrapper class to use with an image loading and caching library.
 * <p>
 * This class was created in order to enable glide to be easily mocked and in the future to easily
 * change the image loading lib.
 */

public class ImageLoaderWrapper {

    public ImageLoaderWrapper() {}

    public void fetchImageToView(Fragment fragment, String url,  ImageView imageView) {
        GlideApp.with(fragment).load(url).centerCrop().into(imageView);
    }
}

File: app/src/main/java/com/juawapps/newsspread/image/ImageLoaderWrapper.kt
package com.juawapps.newsspread.image

import android.support.v4.app.Fragment
import android.widget.ImageView

/**
 * This is a wrapper class to use with an image loading and caching library.
 *
 *
 * This class was created in order to enable glide to be easily mocked and in the future to easily
 * change the image loading lib.
 */

class ImageLoaderWrapper {

    fun fetchImageToView(fragment: Fragment, url: String?, imageView: ImageView) {
        GlideApp.with(fragment).load(url).centerCrop().into(imageView)
    }
}
